### PR TITLE
Add locate functionality

### PR DESCRIPTION
**Pull request by ramindu90**
## Automation tests
 - Unit tests 

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: component/src/main/java/io/siddhi/extension/execution/string/LocateFunctionExtension.java [hunk 1]
```diff
@@ -141,10 +141,16 @@ protected Object execute(Object[] objects, State state) {
         if (objects.length == 3) {
             positionInTheStringToBeginSearch = (Integer) objects[2];
             stringToBeSearched = stringToBeSearched.substring(positionInTheStringToBeginSearch);
+            return stringToBeSearched.indexOf(stringToBeSearchedFor) + positionInTheStringToBeginSearch;
         }
         return stringToBeSearched.indexOf(stringToBeSearchedFor);
     }
 
+    public static void main(String[] args) {
```
Comment (dnwick) [inline component/src/main/java/io/siddhi/extension/execution/string/LocateFunctionExtension.java:149]: we need to remove this ?

File: component/src/main/java/io/siddhi/extension/execution/string/LocateFunctionExtension.java [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+import io.siddhi.core.util.config.ConfigReader;
+import io.siddhi.core.util.snapshot.state.State;
+import io.siddhi.core.util.snapshot.state.StateFactory;
+import io.siddhi.query.api.definition.Attribute;
+
+import static io.siddhi.query.api.definition.Attribute.Type.INT;
+import static io.siddhi.query.api.definition.Attribute.Type.STRING;
+
+/**
+ * locate(string, strin)
```
Comment (AnuGayan) [inline component/src/main/java/io/siddhi/extension/execution/string/LocateFunctionExtension.java:41]: (string, string)

File: component/src/test/java/io/siddhi/extension/execution/string/LocateFunctionExtensionTestCase.java [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.siddhi.extension.execution.string;
+
+import io.siddhi.core.SiddhiAppRuntime;
+import io.siddhi.core.SiddhiManager;
+import io.siddhi.core.event.Event;
+//import io.siddhi.core.exception.SiddhiAppCreationException;
```
Comment (dnwick) [inline component/src/test/java/io/siddhi/extension/execution/string/LocateFunctionExtensionTestCase.java:24]: shall we remove these too

File: component/src/test/java/io/siddhi/extension/execution/string/LocateFunctionExtensionTestCase.java [hunk 4]
```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (c)  2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (AnuGayan) [inline component/src/test/java/io/siddhi/extension/execution/string/LocateFunctionExtensionTestCase.java:2]: 2020??

**Pull request merged**